library/core/src/ptr/alignment.rs

```diff
@@ -370,3 +376,74 @@ enum AlignmentEnum {
     _Align1Shl62 = 1 << 62,
     _Align1Shl63 = 1 << 63,
 }
+
+#[cfg(kani)]
+#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    // pub const fn of<T>() -> Self
+    #[kani::proof]
+    pub fn check_of_i32() {
+        let alignment = Alignment::of::<i32>();
```

Comment:
Why we don't add contracts for `of`? Is this guarantee to return "any" that is power of two?

Reply:
I can neither make a `requires` nor an `ensures` clause work for functions that are generic. I get (with the same error for the below `ensures` clause):
```
error[E0282]: type annotations needed
  --> /verify-rust-std.git/library/core/src/ptr/alignment.rs:51:5
   |
51 |     #[requires(mem::align_of::<T>().is_power_of_two())]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type
   |
   = note: this error originates in the attribute macro `kani_core::requires` (in Nightly builds, run with -Z macro-backtrace for more info)
```
Here are the contracts that I tried:
```
    #[requires(mem::align_of::<T>().is_power_of_two())]
    #[ensures(|result : &Alignment| result.as_usize().is_power_of_two())]
```

Reply:
Hmmm... That shouldn't be the case. Contracts should work with generic arguments

Reply:
Which version of Kani are you using?

Reply:
Yay, works just fine with the updated version of Kani. Contracts added.